In [2]:
import numpy as np

## 1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks): zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110], ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]. Используя математические операции, посчитать коэффициенты линейной регрессии, приняв за X заработную плату (то есть, zp - признак), а за y - значения скорингового балла (то есть, ks - целевая переменная). Произвести расчет как с использованием intercept, так и без.

In [3]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [4]:
zp_mean = np.sum(zp) / len(zp)
ks_mean = np.sum(ks) / len(ks)
ks_zp_mean = np.sum([zp[i] * ks[i] for i in range(len(zp))]) / len(zp)
zp_square_mean = np.sum([zp[i] ** 2 for i in range(len(zp))]) / len(zp)
print(zp_mean, ks_mean, ks_zp_mean, zp_square_mean)

101.4 709.9 81141.7 13776.6


In [7]:
b = (ks_zp_mean - ks_mean * zp_mean) / (zp_square_mean - zp_mean ** 2)
round(b, 4)

2.6205

In [8]:
a = ks_mean - b * zp_mean
round(a, 4)

444.1774

## $ks = 444.1774 + 2.6205 \cdot zp$

## 2. В каких случаях для вычисления доверительных интервалов и проверки статистических гипотез используется таблица значений функции Лапласа, а в каких - таблица критических точек распределения Стьюдента?

### Таблица значений функции Лапласа используется при больших выборках и известной дисперсии. При небольших выборках и неизвестной дисперсии - таблица критических точек распределения Стьюдента.